tests/init/pos/byname.scala

```diff
@@ -0,0 +1,17 @@
+trait T:
+  def bar(i: => Int): Int
+
+class A extends T:
+  override def bar(i: => Int): Int = i + 1
+
+class B extends T:
+  override def bar(i: => Int): Int = i + 2
```

Comment:
Add another test case with something like:

```suggestion
  override def bar(i: => Int): Int = i + bar(i)
```